### PR TITLE
web: show last edited file in statusbar

### PR DESCRIPTION
**Pull request by jazzdan**
Screenshot from 2019-04-25 10-12-54

### CODE REVIEW

File: web/src/mostRecentBuild.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  if (b1Date > b2Date) {
+    return -1
+  }
+  if (b1Date < b2Date) {
+    return 1
+  }
+  return 0
+}
+
+type BuildTuple = {
```
Comment (jazzdan) [inline web/src/mostRecentBuild.ts]: I'm not wild about the name of this type. Open to suggestions but it's not super urgent.

File: web/src/Statusbar.scss [hunk 2]
```diff
@@ -47,3 +47,6 @@
   padding-left: $spacing-unit/2;
   flex-grow: 1;
 }
+.Statusbar-panel--lastEditMessage {
+  color: $color-gray-lightest
```
Comment (yuindustries) [inline web/src/Statusbar.scss]: Missing `;`

File: web/src/mostRecentBuild.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    .map(r => makePendingBuild(r))
+    .filter(b => !isZeroTime(b.since))
+    .sort(buildByDate)
+
+  if (pendingBuildsSorted.length > 0) {
+    return pendingBuildsSorted[0]
+  }
+
+  let buildHistorySorted = resources
+    .flatMap(r => r.buildHistory.map(b => makeBuildHistory(r, b)))
```
Comment (yuindustries) [inline web/src/mostRecentBuild.ts]: Nice! I didn't know about `flatMap`... 🤓

**Comment by jazzdan** [comment]
Going to make another change to make this explode less when there are lots of files in the last edit.

File: web/src/mostRecentBuild.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import { Build } from "./types"
+import { isZeroTime } from "./time"
+
+export type ResourceWithBuilds = {
+  name: string
+  buildHistory: Array<Build>
+  pendingBuildSince: string
```
Comment (nicks) [inline web/src/mostRecentBuild.ts]: i'm surprised this property isn't a Date?

**Pull request merged**